### PR TITLE
[orchagent]: Add ARP update script to maintain VLAN neighbors

### PR DESCRIPTION
**Pull request by stcheng**
- Extend ARP reachable time to 30min
- Add arping to docker-swss
- Add arp_update.sh script to routinely probe neighbors

Signed-off-by: Shuotian Cheng <shuche@microsoft.com>

### CODE REVIEW

**Comment by stcheng** [comment]
⏳

**Comment by pavel-shirshov** [comment]
So you implemented gratuitous ARP here? GARP?
https://en.wikipedia.org/wiki/Address_Resolution_Protocol#ARP_announcements

**Comment by stcheng** [comment]
@pavel-shirshov Yes. Because after the neighbor is created, all the traffic goes through data plane and no packets will be received from the CPU. Thus after certain time the neighbor will be updated to stale state. This is one [reference ](https://support.cumulusnetworks.com/hc/en-us/articles/202012933-Changing-ARP-timers-in-Cumulus-Linux)describing similar issue here.

File: dockers/docker-orchagent/arp_update.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+VLAN=`sonic-cfggen -m /etc/sonic/minigraph.xml -v minigraph_vlan_interfaces`
+if [ "$VLAN" == "[]" ]; then
+    exit 0
+fi
+
+arpingcmd="sed -e 's/ / -i /' -e 's/^/arping -w 0 -c 1 /' -e 's/$/;/'"
+ipcmd="ip -4 neigh show | grep Vlan | cut -d ' ' -f 1,3 | ${arpingcmd}"
```
Comment (stcheng) [inline dockers/docker-orchagent/arp_update.sh]: remove the hardcoded 'vlan' and use for loop

File: dockers/docker-orchagent/arp_update.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
```
Comment (stcheng) [inline dockers/docker-orchagent/arp_update.sh]: add options for timeout
add options for one-time shot

**Comment by stcheng** [comment]
add comments in script

**Comment by stcheng** [comment]
add todo: ipv6

File: dockers/docker-orchagent/arp_update.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+VLAN=`sonic-cfggen -m /etc/sonic/minigraph.xml -v minigraph_vlan_interfaces`
```
Comment (qiluo-msft) [inline dockers/docker-orchagent/arp_update.sh]: >sonic-cfggen -m /etc/sonic/minigraph.xml -v minigraph_vlan_interfaces [](start = 6, length = 69)

Move into config.sh, as other containers.

**Comment by lguohan** [comment]
Can you add description to describe the problem this commit is trying to solve?

**Comment by stcheng** [comment]
updated

File: dockers/docker-orchagent/Dockerfile.j2 [hunk 4]
```diff
@@ -26,6 +26,7 @@ debs/{{ deb }}{{' '}}
 RUN apt-get clean -y; apt-get autoclean -y; apt-get autoremove -y
 RUN rm -rf /debs
 
+COPY ["arp_update", "/usr/bin/"]
 COPY ["start.sh", "orchagent.sh", "swssconfig.sh", "/usr/bin/"]
```
Comment (qiluo-msft) [inline dockers/docker-orchagent/Dockerfile.j2]: Merge into one line.

File: dockers/docker-orchagent/arp_update [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    case $opt in
+        h) usage ;;
+        o) one_time_trigger=true ;;
+        p) periodic_time=$OPTARG ;;
+        :) usage ;;
+        \?) usage ;;
+    esac
+done
+
+while ! $one_time_trigger; do
```
Comment (qiluo-msft) [inline dockers/docker-orchagent/arp_update]: >one_time_trigg [](start = 9, length = 14)

if one_time_trigger == false, nothing happens.

File: dockers/docker-orchagent/supervisord.conf [hunk 6]
```diff
@@ -59,3 +59,9 @@ autorestart=false
 stdout_logfile=syslog
 stderr_logfile=syslog
 
+[program:arp_update]
+command=/usr/bin/arp_update
+autostart=true
```
Comment (qiluo-msft) [inline dockers/docker-orchagent/supervisord.conf]: >true [](start = 10, length = 4)

suggest follow Joe's pattern:
1. /usr/bin/start.sh is the only section autostart
2. /usr/bin/start.sh will start all other sections, so the order is guaranteed

File: dockers/docker-orchagent/arp_update (hunk 5 above)
Comment (stcheng) [inline dockers/docker-orchagent/arp_update]: fixed

**Comment by stcheng** [comment]
@qiluo-msft updated

File: dockers/docker-orchagent/supervisord.conf (hunk 6 above)
Comment (stcheng) [inline dockers/docker-orchagent/supervisord.conf]: fixed

File: dockers/docker-orchagent/Dockerfile.j2 (hunk 4 above)
Comment (stcheng) [inline dockers/docker-orchagent/Dockerfile.j2]: fixed

**Comment by stcheng** [comment]
@qiluo-msft updated the commit using supervisord to restart the process. right now no exit code is needed.

**Comment by stcheng** [comment]
@lguohan please review again.

**Pull request merged**